.github/actions/build-external/action.yml

```diff
@@ -22,11 +22,11 @@ runs:
       id: cache-check
       with:
         path: build/external/${{ inputs.library }}-${{ inputs.version }}
-        key: ${{ inputs.library }}-${{ inputs.version }}-${{ inputs.os }}
+        key: ${{ inputs.library }}-${{ inputs.version }}-${{ runner.os }}-${{ matrix.build-output }}
```

Comment:
Does this action inherit `matrix` from the parent context? I thought we only got the variables that we explicitly declared as inputs.